File: algorithms/flatland-space-station/flatland_space_station_v2.py
def flatlandSpaceStation(n, c):
  if (n == len(c)):
    return 0
  
  noSpaceStation = []
  prevNearestSpaceStation = min(c)
  maxD = 0
  
  for i in range(n):
    if i not in c:
      noSpaceStation.append(i)
      continue
      
    for city in noSpaceStation:
      tmp = [abs(city - prevNearestSpaceStation), abs(city - i)]
      nearestDistance = min(tmp)
      if (nearestDistance > maxD):
        maxD = nearestDistance
      
    prevNearestSpaceStation = i
    noSpaceStation = []
      
  nearestDistance = max(noSpaceStation) - prevNearestSpaceStation
  if (nearestDistance > maxD):
    maxD = nearestDistance            

  return maxD

File: algorithms/flatland-space-station/flatlandSpaceStationV2.php
<?php
function flatlandSpaceStation($n, $c) {
  if ($n == count($c)) {
      return 0;
  }

  sort($c);
  $maxD = min($c);
  $tmp = 0;

  foreach ($c as $spaceStation) {
    $middleCity = intdiv(($spaceStation + $tmp), 2);
    $distance = min($middleCity - $tmp, $spaceStation - $middleCity);
    
    if ($distance > $maxD) {
      $maxD = $spaceStation;
    }
    
    $tmp = $spaceStation;
  };

  if (($n - 1) - $spaceStation > $maxD) {
    $maxD = ($n - 1) - $spaceStation;
  }

  return $maxD;
}
?>
